docs/dev/src/modules/ROOT/partials/eventsourced-snapshots.adoc

```diff
@@ -0,0 +1,11 @@
+Snapshots are an important optimization for Event Sourced Entities that emit many events. Rather than reading the entire journal upon loading or restart, Kalix can initiate them from a snapshot.
+
+Snapshots are stored and handled automatically by Kalix without any specific code required. Snapshots are stored after a configured number of events:
+
+[source,conf,indent=0]
+.src/main/resources/application.conf
+----
+include::java:example$spring-eventsourced-shopping-cart/src/main/resources/application.conf[tag=snapshot-every]
+----
+
+When the Event Sourced Entity is loaded again, the snapshot will be loaded before any other events are received.
```

Comment:
This would be duplicated between java/scala and spring so decided to move to a partial.. (we should probably agree if this is a strategy we want to follow, or if this shared partials should be somewhere else). I've tried to add it to the "developing" module but it did not work.. ended-up adding to the ROOT. but let me know what would be best here..